executor/common_linux.h

```diff
@@ -1165,18 +1175,21 @@ static long syz_mount_image(volatile long fsarg, volatile long dir, volatile uns
 		// and if two parallel executors mounts fs with the same uuid, second mount fails.
 		strcat(opts, ",nouuid");
 	}
-	debug("syz_mount_image: size=%llu segs=%llu loop='%s' dir='%s' fs='%s' flags=%llu opts='%s'\n", (uint64)size, (uint64)nsegs, loopname, (char*)dir, fs, (uint64)flags, opts);
+	dir2 = mkdtemp(dirtemplate);
+	debug("syz_mount_image: size=%llu segs=%llu loop='%s' dir2='%s' fs='%s' flags=%llu opts='%s'\n", (uint64)size, (uint64)nsegs, loopname, (char*)dir2, fs, (uint64)flags, opts);
 #if SYZ_EXECUTOR
-	cover_reset(0);
+	// TODO: cover->reset seems to set cov->data to 0. If the below call is enabled, no coverage
+	// seems to be collected from mount(). Is the intention to measure coverage of mount alone?
+	// If so, should there be another call to set cov->data to another mmap'd region?
+	// cover_reset(0);
```

Comment:
Should cov->data point to another mmap'd region?

Reply:
This call does not set `cov->data` to 0. It sets `*(uint64*)cov->data`, which is an 8-byte location `cov->data` points to.
Yes, the intention is to collect coverage only from the mount call. So we need this.
Unless I am overlooking something obvious here, there must be some other reason why you don't see coverage with this uncommented.

Reply:
When I uncommented this call and run the program with syz-execprog I do see the coverage:
```
#0 [471ms] -> syz_mount_image$ext4(0x20000000, 0x20000100, 0x6400000, 0x12d, 0x20000200, 0x0, 0x20024b00)
syz_mount_image: size=135266304 segs=301 loop='/dev/loop1505335290' dir2='/tmp/template.Nsl5th' fs='ext4' flags=0 opts=',errors=continue'
[   97.370702] EXT4-fs (loop1505335290): mounted filesystem with ordered data mode. Opts: ,errors=continue
#0 [485ms] <- syz_mount_image$ext4=0x0 errno=0 cover=81016 
```

With cover_reset commented out it collect much more coverage:
```
#0 [700ms] <- syz_mount_image$ext4=0x0 errno=0 cover=262143 
```
Which is 256K which means that the trace buffer has overflowed and we lost coverage from mount.
So cover_reset is very much on purpose here.

Reply:
Ah yes, careless mistake with the "cov->data = 0" when skimming through the code. :(
I see, so the trace buffer is getting overflown if not reset, thanks. I'll change this back.